src/dscl.ts

```diff
@@ -0,0 +1,316 @@
+// https://stackoverflow.com/a/70994696/6183068
+const satisfies =
+  <T>() =>
+  <U extends T>(u: U) =>
+    u;
+
+const generateDsclPath: Fig.Generator = {
+  trigger: "/",
+  getQueryTerm: "/",
+  custom: async (tokens, executeShellCommand) => {
+    const lastToken = tokens[tokens.length - 1];
+    if (lastToken === "") {
+      return [];
+    }
+    // can't guarantee that dscl is the first token. If it's not found,
+    // this is -1, so searching for `datasource` will start from index 0
+    const dsclIndex = tokens.findIndex((token) => token === "dscl");
+    let datasource: string;
+    for (let i = dsclIndex + 1; i < tokens.length; i++) {
+      const token = tokens[i];
+      // skip if the current token is a flag
+      if (token.startsWith("-")) {
+        continue;
+      }
+      // skip if the previous token is a flag that takes an argument
+      const prev = tokens[i - 1];
+      if (
+        dsclOptions.find(
+          (option) => [option.name].flat().includes(prev) && !!option.args
+        )
+      ) {
+        continue;
+      }
+      datasource = token;
+      break;
+    }
+    const lastSlashIndex = lastToken.lastIndexOf("/");
+    const path = lastToken.slice(0, lastSlashIndex < 0 ? 0 : lastSlashIndex);
+    const command = `dscl ${datasource} -list ${path || "/"}`;
+    const lines = await executeShellCommand(command);
+    return lines
+      .trim()
+      .split("\n")
+      .map((line) => ({
+        name: line,
+        icon: "📁",
+        priority: line.slice(line.lastIndexOf("/") + 1).startsWith("_")
```

Comment:
Just realized the cleaner+faster way of doing this 😄
Ultimately doesn't _really_ matter but it's still perf increase
```suggestion
        priority: line[line.lastIndexOf("/") + 1] === "_"
```